server/routes/playlist.js

```diff
@@ -101,4 +113,30 @@ router.get('/get-track-vector', jwt.verifyJWT, async (req, res, next) => {
   }
 })
 
+/* adds playlist as a favorite playlist */
+router.post('/add-favorite', jwt.verifyJWT, async (req, res, next) => {
+  try {
+    const userId = req.userId
+    const { playlistId } = req.body
+
+    await Playlist.findOneAndUpdate({ userId, playlistId }, { favorite: true })
```

Comment:
Do you mean if somehow the playlist was deleted from the database, because the only way for now to remove/add a playlist is if it was in the database already.